frontend/src/components/JobCard.tsx

```diff
@@ -1,23 +1,56 @@
-import {Card, CardActions, CardContent, Typography} from "@mui/material";
+import {Button, Card, CardActions, CardContent, Dialog, DialogActions, DialogTitle, Typography} from "@mui/material";
 import {Job} from "../model/Job";
-import React from "react";
+import React, {useState} from "react";
 import {Link} from "react-router-dom";
+import DeleteForeverIcon from "@mui/icons-material/DeleteForever";
 
 type JobCardProps = {
     job: Job
+    deleteJobById: (id: string | undefined) => void
 }
 
 export default function JobCard(props: JobCardProps) {
+    const [open, setOpen] = useState(false)
```

Comment:
Maybe you could give the constant a clearer name, for example `dialogBeforeDeleteIsOpen`